doc/languages-frameworks/agda.section.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Agda
+author: Alex Rice (alexarice)
+date: 2020-01-06
+---
+# Agda
+
+## How to use Agda
+
+Agda can be installed with `agda.Agda`. For example:
+```
+$ nix-env -iA agda.Agda
+```
+
+To use agda with libraries, the `withPackages` function can be used. This function either takes:
++ A list of packages,
++ or a function which returns a list of packages when given the `agda` attribute set.
+
+For example, suppose we wanted a version of agda which has access to the standard library. This can be obtained with the expressions:
+
+```
+agda.withPackages [ agda.standard-library ]
+```
+
+or
+
+```
+agda.withPackages (p: [ p.standard-library ])
+```
+
+If you want to use a library in your home directory (for instance if it doesn't have a nix derivation or is a development version) then an attribute set can be passed into `agda.withPackages` with the following arguments:
+
+```
+agda.withPackages {
+  pkgs = [ agda.standard-library ];
+  homeLibraries = "path/to/my/package.agda-lib";
+}
+```
+multiple libraries can be passed in by separating them with newlines.
+
+Agda will not by default use these libraries. To tell agda to use the library we have some options:
+- Call `agda` with the library flag:
+```
+$ agda -l standard-library -i . MyFile.agda
+```
+- Write a `.agda-lib` file for the project you are working on which may look like:
+```
+name: my-library
+include: .
+depends: standard-library
+```
+- Create the file `~/.agda/defaults` and add any libraries you want to use by default.
+
+More information can be found [here](https://agda.readthedocs.io/en/v2.6.0.1/tools/package-system.html).
+
+## Compiling Agda
+Agda modules can be compiled with the `--compile` flag. A version of `ghc` with `ieee` is made available to the Agda program via the `--with-compiler` flag. This can be overriden by a different version of `ghc` as follows:
```

Comment:
```suggestion
Agda modules can be compiled with the `--compile` flag. A version of `ghc` with `ieee` is made available to the Agda program via the `--with-compiler` flag.
This can be overridden by a different version of `ghc` as follows:
```